[Export] Introduce class_fqn into CustomObjArgument

SherlockNoMad opened the pull request:
Summary:
Class FQN is needed when unpacking CustomObj instance.
For all other Arguments, e.g. Tensor, TensorList, SymInt, we always know their exact type. However, CustomObjArgument had an opaque type.
Adding this field also helps unveiling the type of this opaque object.

Test Plan: CI

Differential Revision: D53029847

zhxchen17 commented on torch/export/graph_signature.py:
```diff
@@ -31,6 +31,7 @@ class SymIntArgument:
 @dataclasses.dataclass
 class CustomObjArgument:
     name: str
+    class_fqn: str
```
is this fqn defined in the context of Python? If so do we have requirements for the backends to properly decode this fqn?

SherlockNoMad replied:
No, this is not tied to python. 

In C++, we are also relying on this fqn to get handle to the CustomClassHolder. 

When user implements a CustomClassHolder class, framework will assign a fqn for this class.

zhxchen17 commented:
@SherlockNoMad yo linters

SherlockNoMad committed:
[Export] Introduce class_fqn into CustomObjArgument (#118158)

Summary:

Class FQN is needed when unpacking CustomObj instance.
For all other Arguments, e.g. Tensor, TensorList, SymInt, we always know their exact type. However, CustomObjArgument had an opaque type.
Adding this field also helps unveiling the type of this opaque object.

This is OSS-only change split from https://www.internalfb.com/diff/D52432583

Test Plan: CI

Reviewed By: zhxchen17

Differential Revision: D53029847
